moto/ec2/models/instances.py

```diff
@@ -595,16 +595,18 @@ def get_cfn_attribute(self, attribute_name: str) -> Any:
 
     def applies(self, filters: List[Dict[str, Any]]) -> bool:
         if filters:
-            applicable = False
+            # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/describe_instances.html
+            # Filters Section in the boto3 documentation
+            # If you specify multiple filters, the filters are joined with an AND, and the request returns only results that match all of the specified filters.
```

Comment:
Direct quote from AWS is great - that immediately explains why it's implemented this way :+1: